ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -112,10 +124,56 @@ let networks path (list: string -> string list) =
          | Some pair -> pair :: acc
       ) [] (list path)
   in
-  path
-  |> find_eths
-  |> List.map (fun (path, prefix) -> find_all_ips path prefix)
-  |> List.concat
+  let find_vifs vif_path =
+    let extract_vif acc vif_id = ((extend vif_path vif_id), vif_id) :: acc in  
+    List.fold_left extract_vif [] (list vif_path)
+  in
+  let compare a b = 
+    let open Pervasives in
+    compare (int_of_string a) (int_of_string b) 
```

Comment:
`int_of_string` can fail and the exception is not caught here. I now realise that you catch the exception below at the call site - maybe add a comment here. To avoid any confusion with the global `compare` function in the code below, I probably would call it something else, like `cmp`.